### PR TITLE
Extra caveat around state in Java EventSourcedBehavior 

### PR DESCRIPTION
**Pull request by johanandren**
Refs #25698

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/persistence.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Java
 
 The first important thing to notice is the `Behavior` of a persistent actor is typed to the type of the `Command`
 because this is the type of message a persistent actor should receive. In Akka Typed this is now enforced by the type system.
-The event and state are only used internally.
+The event and state are only used internally in the `EventSourcedBehavior`.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/typed/persistence.md]: Maybe that sentence is just adding confusion. Events can be consumed by read-side. State may be sent as a message if one like (although I wouldn't recommend it).

File: akka-docs/src/main/paradox/typed/persistence.md [hunk 2]
```diff
@@ -48,6 +48,14 @@ The components that make up a EventSourcedBehavior are:
 * `commandHandler` defines how to handle command by producing Effects e.g. persisting events, stopping the persistent actor.
 * `eventHandler` returns the new state given the current state when an event has been persisted.
 
+@@@ div { .group-java }
+
+Note that the concrete class does not contain any fields with state like a regular POJO. All state of the 
+`EventSourcedBehavior` must be represented in the `State` or else they will not be persisted and therefore be
+lost when the actor is stopped or restarted.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/typed/persistence.md]: Maybe add: Updates to the `State` are always performed in the `eventHandler` based on the events.

File: akka-docs/src/main/paradox/typed/persistence.md (hunk 1 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/typed/persistence.md]: Yeah, I was on the fence. Let's remove it.

Comment (helena) [reply akka-docs/src/main/paradox/typed/persistence.md]: +1

**Review by helena — APPROVED** [review]
LGTM

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**